packages/block-library/src/gallery/edit.js

```diff
@@ -377,10 +416,41 @@ class GalleryEdit extends Component {
 	}
 }
 export default compose( [
-	withSelect( ( select ) => {
+	withSelect( ( select, { attributes: { ids } } ) => {
+		const { getMedia } = select( 'core' );
 		const { getSettings } = select( 'core/block-editor' );
-		const { mediaUpload } = getSettings();
-		return { mediaUpload };
+		const {
+			imageSizes,
+			mediaUpload,
+		} = getSettings();
+
+		const resizedImages = reduce(
+			ids,
+			( currentResizedImages, id ) => {
+				if ( ! id ) {
+					return currentResizedImages;
+				}
+				const image = getMedia( id );
```

Comment:
This is marking an HTTP request per image, if a post as multiple galleries with multiple images we may end up with issue https://github.com/WordPress/gutenberg/issues/17465. I guess we could apply a similar solution to https://github.com/WordPress/gutenberg/pull/17504.

Reply:
Done in 231da67